osu.Game/Screens/Edit/Compose/Components/ComposeBlueprintContainer.cs

```diff
@@ -321,21 +324,24 @@ protected override void Update()
                 {
                     case PlacementBlueprint.PlacementState.Waiting:
                         if (!Composer.CursorInPlacementArea)
-                            removePlacement();
+                            CurrentPlacement.Hide();
+                        else if (Composer.CursorInPlacementArea)
+                            CurrentPlacement.Show();
```

Comment:
??????????????????????????????????????

Reply:
So is the direction okay or not

Reply:
Bit of a surprising reaction since `PlacementBlueprint`s were actually historically like this, until it was "simplified" at some point to not have this.

Reply:
read the code you're written again, please.

Reply:
addressed in https://github.com/ppy/osu/pull/30411/commits/0e65655feffcbe38f6e9b9573e16227c7225bbc3 :-)